testutil/filefixtures/types.go

```diff
@@ -0,0 +1,184 @@
+package filefixture
+
+import (
+	"archive/tar"
+	"bytes"
+	"fmt"
+	"io/ioutil"
+	"sort"
+	"strings"
+	"time"
+
+	"polydawn.net/repeatr/lib/fs"
+)
+
+type FixtureFile struct {
+	Metadata fs.Metadata
+	Body     []byte
+}
+
+type Fixture struct {
+	Name  string
+	Files []FixtureFile
+}
+
+type ComparisonOptions uint32
+
+const (
+	ComparePerms = ComparisonOptions(0001)
+	CompareMtime = ComparisonOptions(0002)
+	CompareAtime = ComparisonOptions(0004)
+	CompareUid   = ComparisonOptions(0010)
+	CompareGid   = ComparisonOptions(0020)
+	CompareSize  = ComparisonOptions(0040)
+	CompareBody  = ComparisonOptions(0100)
+
+	CompareDefaults = ComparePerms | CompareMtime | CompareUid | CompareGid |
+		CompareSize | CompareBody
+	CompareAll = CompareDefaults | CompareAtime
+)
+
+type filesByPath []FixtureFile
+
+func (a filesByPath) Len() int           { return len(a) }
+func (a filesByPath) Swap(i, j int)      { a[i], a[j] = a[j], a[i] }
+func (a filesByPath) Less(i, j int) bool { return a[i].Metadata.Name < a[j].Metadata.Name }
+
+func (f Fixture) defaults() Fixture {
+	for i, ff := range f.Files {
+		f.Files[i] = defaults(ff)
+	}
+	return f
+}
+
+func defaults(f FixtureFile) FixtureFile {
+	if f.Metadata.Typeflag == '\x00' {
+		if f.Body == nil {
+			f.Metadata.Typeflag = tar.TypeDir
+		} else {
+			f.Metadata.Typeflag = tar.TypeReg
+		}
+	}
+	if f.Metadata.Mode == 0 {
+		switch f.Metadata.Typeflag {
+		case tar.TypeDir:
+			f.Metadata.Mode = 0755
+		default:
+			f.Metadata.Mode = 0644
+		}
+	}
+	if f.Metadata.Size == 0 {
+		f.Metadata.Size = int64(len(f.Body))
+	}
+	if f.Metadata.Uid == 0 {
+		f.Metadata.Uid = 10000
+	}
+	if f.Metadata.Gid == 0 {
+		f.Metadata.Gid = 10000
+	}
+	if f.Metadata.ModTime.IsZero() {
+		f.Metadata.ModTime = time.Unix(0, 0).UTC()
+	}
+	if f.Metadata.AccessTime.IsZero() {
+		f.Metadata.AccessTime = time.Unix(0, 0).UTC()
+	}
+	return f
+}
+
+/*
+	Create files described by the fixtures on the real filesystem path given.
+*/
+func (ffs Fixture) Create(basePath string) {
+	for _, f := range ffs.Files {
+		fs.PlaceFile(basePath, f.Metadata, bytes.NewBuffer(f.Body))
+	}
+	// re-do time enforcement... in reverse order, so we cover our own tracks
```

Comment:
Am I understanding you correctly in that "cover our own tracks" means preventing messing up a directory?
Either way, could you clarify this comment please.